optaplanner-core/src/main/java/org/optaplanner/core/api/score/Score.java

```diff
@@ -60,15 +63,22 @@
      * For example {@code -7init/0hard/-8soft} returns {@code 0hard/-8soft}.
      *
      * @return equal score except that {@link #getInitScore()} is {@code 0}.
+     * @deprecated Use {@link #withInitScore(int)} with 0 as the argument.
      */
-    Score_ toInitializedScore();
+    @Deprecated
+    default Score_ toInitializedScore() {
+        if (isSolutionInitialized()) {
+            return (Score_) this;
+        } else {
+            return withInitScore(0);
+        }
+    }
 
     /**
      * For example {@code 0hard/-8soft} with {@code -7} returns {@code -7init/0hard/-8soft}.
      *
      * @param newInitScore always negative (except in statistical calculations), 0 if all planning variables are initialized
      * @return equals score except that {@link #getInitScore()} is set to {@code newInitScore}
-     * @throws IllegalStateException if the original {@link #getInitScore()} is not 0
```

Comment:
Technically this is a backwards incompatible change. Used to throw, doesn't anymore. I think it should be fine.

Reply:
EasyScoreDirector and IncrementalScoreDirector should fail fast if the EasyScoreCalculator etc return a Score with a non zero initScore. Before they did, now they don't I believe.

Reply:
I'd add a fail-fast check in both those directors, if the score returned by calculateScore() has a non-zero initScore.